airbyte-server/src/main/java/io/airbyte/server/handlers/OAuthHandler.java

```diff
@@ -342,10 +343,17 @@ JsonNode getOauthFromDBIfNeeded(final JsonNode oAuthInputConfigurationFromDB, fi
 
   @VisibleForTesting
   JsonNode getOAuthInputConfiguration(final JsonNode hydratedSourceConnectionConfiguration, final Map<String, String> pathsToGet) {
-    return Jsons.jsonNode(pathsToGet.entrySet().stream()
-        .collect(Collectors.toMap(
-            Map.Entry::getKey,
-            entry -> JsonPaths.getSingleValue(hydratedSourceConnectionConfiguration, entry.getValue()).get())));
+    final Map<String, JsonNode> result = new HashMap<>();
+    pathsToGet.forEach((k, v) -> {
+      final Optional<JsonNode> configValue = JsonPaths.getSingleValue(hydratedSourceConnectionConfiguration, v);
```

Comment:
Could we instead make sure the `hydratedSourceConnectionConfiguration` has all the expected paths?

Reply:
The expected behavior is actually that `hydratedSourceConnectionConfiguration` _doesn't_ have the path we're trying to get in this case.

Because of the way the spec is defined, `credentials.subdomain` is only present in the config if the OAuth authentication type has been selected (since it's only used in the oauth case)

So, if you're coming from an API key auth, your config will look something like `{"credentials": {"api_token": "sometoken", "auth_type": "api_token"}}`. When this method runs after switching for oauth, we try to get the value of `credentials.subdomain` from the existing configuration but realize it's not there. Previously since we always expected it to be present an error would be thrown. Now, we just log a warning and don't retrieve the value.